stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -198,12 +198,17 @@ public class MessageListViewModel @JvmOverloads constructor(
     }
 
     private fun onThreadModeEntered(parentMessage: Message) {
-        currentMode = Mode.Thread(parentMessage)
         val parentId: String = parentMessage.id
-        val threadController = domain.useCases.getThread(cid, parentId).execute().data()
-        threadMessages = threadController.messages
-        setThreadMessages(threadMessages)
-        domain.useCases.threadLoadMore(cid, parentId, MESSAGES_LIMIT).enqueue()
+        domain.useCases.getThread(cid, parentId).enqueue { threadControllerResult ->
+            if (threadControllerResult.isSuccess) {
+                threadControllerResult.data().run {
+                    currentMode = Mode.Thread(parentMessage)
+                    threadMessages = messages
+                    setThreadMessages(threadMessages)
+                    domain.useCases.threadLoadMore(cid, parentId, MESSAGES_LIMIT).enqueue()
+                }
+            }
```

Comment:
I had to stare at this for a bit to figure out what `run` does here, what references within it come from the `data()` that it was called on (I think it's just `messages`, and nothing else?). Seems a bit unnecessary.

Reply:
Updated ;)

Reply:
Readability win of the day - thank you! 😄